### ISSUE TITLE
`CrossValCurate.fit_transform` adds spurious column `row_id`to input data 

### ISSUE THREAD

**Issue by sumanthprabhu** [sumanthprabhu/DQC-Toolkit#5]
**Describe the bug**
`CrossValCurate.fit_transform` method seems to be adding an additional column `row_id` to the input data. This is unexpected behaviour since a separate modified pandas dataframe containing the expected results is returned for downstream consumption.

**To Reproduce**
``` py
from datasets import load_dataset
from dqc import CrossValCurate
import pandas as pd 

dataset = 'ag_news'
dset = load_dataset(dataset)
train_data = pd.DataFrame(dset['train'])

cvc = CrossValCurate(random_state=seed)

train_data_modified = cvc.fit_transform(train_data, y_col_name='label')
print(train_data.columns)
```

**Expected behavior**
Index(['text', 'label'], dtype='object')

**Actual results**
Index(['text', 'label', 'row_id'], dtype='object')

**System Information:**
'os_type': 'Linux',
 'os_version': '5.15.133+',
 'python_version': '3.10.13',
 'dqc-toolkit': '0.1.0',
 'transformers': '4.39.3',
 'sentence_transformers': '2.7.0',
 'scikit-learn': '1.4.2'

**Additional context**
